src/librustc_mir/borrow_check/invalidation.rs

```diff
@@ -183,6 +183,29 @@ impl<'cx, 'tcx> Visitor<'tcx> for InvalidationGenerator<'cx, 'tcx> {
                     }
                 }
             }
+            TerminatorKind::InlineAsm { template: _, ref operands, options: _, destination: _ } => {
+                for op in operands {
+                    match op {
+                        InlineAsmOperand::In { reg: _, value }
+                        | InlineAsmOperand::Const { value } => {
+                            self.consume_operand(location, value);
+                        }
+                        InlineAsmOperand::Out { reg: _, late: _, place, .. } => {
+                            if let Some(place) = place {
+                                self.mutate_place(location, place, Shallow(None), JustWrite);
```

Comment:
Similarly here `JustWrite` sounds wrong if its not `lateout`. Applies to inout too.